### DISCUSSION TITLE
Genesis Adapters and Random Infrequent Phantom Button Presses

### DISCUSSION THREAD

**Discussion by ca7bc0c5f2** [darthcloud/BlueRetro#1040] [Q&A]
I've built two Genesis adapters. One with Multi-tap support (which I am yet to test). I utilized the AIO adapters from pmgducati. I've tried several different Bluetooth controllers (Dualshock 4, Duelsense, Duelsense Edge) and this same issue persists. When playing Streets of Rage or SoR 2 the game will get a random start press or A press. This manifests as a special spend that's unintended or an unintentional pause. This console has worked OK previously with first party controllers, so I don't think that is the issue. I am using a power supply that should provide enough power for the attached BlueRetro and MegaSD I am playing off of. I also have TurboGrafx16, OD style 12 input arcade adapters, and PS2 adapters that all seem to work correctly, granted those have been less frequently used/less strenuously tested. One additional test I tried was to pause the game and see if it unpaused after some time. It never spontaneously unpaused. The phantom inputs seem to only happen after a rapid succession of actual inputs. Flipping P2 and P1 or using only one of the player connectors seems to have not helped either. Auto console detection on the universal firmware has been the setup approach when using the Genesis. I have not tried manually defining it. Using BlueRetro.io I did determine that detection appears to be accurate. I have reviewed and re-soldered connections several times, and alcohol brushed flux off the boards. 

Any thoughts on why this could be occurring? I'm happy to submit a bug report if that is the better channel for this. If it is, could I ask for guidance on how to obtain a Bluetooth trace (a link would be fine, I haven't been able to find a clear guide)?  Thanks in advance!

**Comment by darthcloud** [comment]
Genesis is tricky and polling pattern vary gretly between games. The ESP32 is quite slow GPIO-wise and making all games work well is hard. You could file an issue but that might take time for me to get to it.
